addons/hr_holidays/models/res_company.py

```diff
@@ -0,0 +1,37 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import date, timedelta
+
+from odoo import fields, models
+
+
+class res_company(models.Model):
+    _inherit = 'res.company'
+
+    leave_type_excess_reset_month = fields.Selection([
+        ('1', 'January'),
+        ('2', 'February'),
+        ('3', 'March'),
+        ('4', 'April'),
+        ('5', 'May'),
+        ('6', 'June'),
+        ('7', 'July'),
+        ('8', 'August'),
+        ('9', 'September'),
+        ('10', 'October'),
+        ('11', 'November'),
+        ('12', 'December'),
+    ], required=True, default='1')
+    leave_type_excess_reset_day = fields.Selection([
+        (str(i + 1), str(i + 1)) for i in range(31)
+    ], default='1')
+
+    def _get_leave_type_reset_day(self, target_date=fields.Date.today):
+        self.ensure_one()
+        year = target_date.year
+        month = int(self.leave_type_excess_reset_month)
+        day = int(self.leave_type_excess_reset_day)
+        leave_type_reset_date = date(year, month, day)
+        if leave_type_reset_date <= target_date:
+            return leave_type_reset_date
+        return leave_type_reset_date + timedelta(years=1)
```

Comment:
If it's smaller return leave_type_reset_date, but if it's larger add another year? Shouldn't this be minus timedelta(years=1), as I think this method should return the last time the leave type was reset right?

Reply:
Indeed! Though after discussion with Geoffrey, that date system is dropped